### PR TITLE
Cleanup .gitignore file

### PR DESCRIPTION
**Pull request by shulard**
I am targetting this branch, because this is a pedantic branch.

Related to: sonata-project/dev-kit#211

## Subject

Remove plateform specific ignore entries from .gitignore file.

### CODE REVIEW

**Comment by greg0ire** [comment]
Thanks @shulard

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
-Resources/doc/_build/*
+/build
+/Resources/doc/_build
+/vendor
 composer.lock
-vendor
-/.php_cs.cache
+phpunit.xml
+.php_cs.cache
```
Comment (core23) [inline .gitignore]: Please sort the files as well like in https://github.com/sonata-project/SonataUserBundle/blob/3.x/.gitignore

Comment (shulard) [inline .gitignore]: This PR was merged before the file was properly sorted... Do I need to open a new PR or can you revert this one ?

**Comment by dbu** [comment]
i would just do a new pull request with the additional cleanup.
reverting does not make a nicer git history. once a commit is in master,
we can't make it invisible without creating a huge mess...

thanks, and don't worry, no harm done.

**Pull request merged**